src/components/SelectionList/Search/ReportListItem.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import {View} from 'react-native';
+import Button from '@components/Button';
+import BaseListItem from '@components/SelectionList/BaseListItem';
+import type {ListItem, ReportListItemProps, ReportListItemType, TransactionListItemType} from '@components/SelectionList/types';
+import Text from '@components/Text';
+import TextWithTooltip from '@components/TextWithTooltip';
+import useLocalize from '@hooks/useLocalize';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import * as CurrencyUtils from '@libs/CurrencyUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import {getSearchParams} from '@libs/SearchUtils';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import ExpenseItemHeaderNarrow from './ExpenseItemHeaderNarrow';
+import TransactionListItem from './TransactionListItem';
+import TransactionListItemRow from './TransactionListItemRow';
+
+function ReportListItem<TItem extends ListItem>({
+    item,
+    isFocused,
+    showTooltip,
+    isDisabled,
+    canSelectMultiple,
+    onSelectRow,
+    onDismissError,
+    shouldPreventDefaultFocusOnSelectRow,
+    onFocus,
+    shouldSyncFocus,
+}: ReportListItemProps<TItem>) {
+    const reportItem = item as unknown as ReportListItemType;
+
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const {isLargeScreenWidth} = useWindowDimensions();
+    const StyleUtils = useStyleUtils();
+
+    const listItemPressableStyle = [styles.selectionListPressableItemWrapper, styles.pv3, item.isSelected && styles.activeComponentBG, isFocused && styles.sidebarLinkActive];
+
+    const handleOnButtonPress = () => {
+        onSelectRow(item);
+    };
+
+    const openReportInRHP = (transactionItem: TransactionListItemType) => {
+        const searchParams = getSearchParams();
+        const currentQuery = searchParams && `query` in searchParams ? (searchParams?.query as string) : CONST.TAB_SEARCH.ALL;
```

Comment:
We should use optional chaining  instead of `query in searchParams`

Reply:
I think we should pass `query` props from src/components/Search.tsx instead of using a new function `getSearchParams`

Reply:
`getSearchParams` has been added to avoid prop drilling, this change would require passing the same value to all items in the list and would introduce many changes.